pkg/util/util_test.go

```diff
@@ -0,0 +1,42 @@
+package util
+
+import (
+	"testing"
+	"time"
+)
+
+const (
+	testAttempts = 1
+)
+
+func TestValidateFetchReadURL(t *testing.T) {
+	tests := []struct {
+		url     string
+		succeed bool
+	}{
+		{"http://example.com", true},
+		{"invalid.url.^&*!@#", false},
+	}
+
+	for _, tt := range tests {
+		_, err := ValidateFetchReadURL(tt.url, testAttempts)
```

Comment:
We are moving to using Golang subtests. Please use t.Run as in https://github.com/tnozicka/opencompose/blob/a21430158830bff240a8252aa4b609e994907238/pkg/encoding/v1/encoding_test.go#L207

You can find out more on https://blog.golang.org/subtests